test/internal/flag/flag.go

```diff
@@ -12,69 +12,72 @@ import (
 
 const FeatureFlagName = "feature"
 
-func NewFeatureFlag(all []string) *StringSlice {
-	return NewStringSlice(FeatureFlagName,
-		"the sets of functionality to test; can be set multiple times, or separated with commas. "+
-			"Supported features: "+strings.Join(all, ", "),
+func NewFeatureFlag(all []string) *StringSet {
+	return NewStringSet(FeatureFlagName,
+		"set of `features` to test; can be set multiple times, with a comma-separated list, or both "+
+			"(supported features: "+strings.Join(all, ", ")+")",
+		false,
 	)
 }
 
-// StringSlice is a type to be used with the standard library's flag.Var
-// function as a custom flag value, similar to "github.com/urfave/cli".StringSlice.
+// StringSet is a type to be used with the standard library's flag.Var
+// function as a custom flag value, similar to "github.com/urfave/cli".StringSet,
+// but it only tracks unique instances.
 // It takes either a comma-separated list of strings, or repeated invocations.
-type StringSlice struct {
-	S StringSet
+type StringSet struct {
+	s  map[string]struct{}
+	cs bool
```

Comment:
nit: can we clarify that this is "case sensitive"? For a second I thought it stood for "comma separated" :D

Reply:
added comment